src/main/java/com/google/devtools/build/lib/syntax/FuncallExpression.java

```diff
@@ -841,12 +842,15 @@ private void evalArguments(
       if (arg.isPositional()) {
         posargs.add(value);
       } else if (arg.isStar()) { // expand the starArg
-        if (!(value instanceof Iterable)) {
+        Collection<?> items;
+        try {
+          items = EvalUtils.toCollection(value, getLocation(), env);
```

Comment:
toCollection allocates a copy when all we need is to iterate over the existing value without retaining it. Is there a variant of toCollection that returns an iterable? If not, it may be time to add one.